packages/ui/react-ui-editor/src/extensions/markdown/toolbar.tsx

```diff
@@ -0,0 +1,51 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { type Extension } from '@codemirror/state';
+import { showTooltip, type TooltipView, type EditorView } from '@codemirror/view';
+import React from 'react';
+import { createRoot } from 'react-dom/client';
+
+import { getFormatting } from './formatting';
+import { Toolbar } from '../../components/Toolbar/Toolbar';
```

Comment:
we might be out of sync here.

first, I don't think extensions should import React code; instead if needed pass in a renderer to the constructor (as with `link`) since the app will bring in it's own rendering opinions, classes, etc. and how it wants to configure the Toolbar -- e.g., extended buttons, menu, etc.

second, for now i think we'll opt to have a fixed toolbar set-up by the app (as with the storybook) rather then a tooltip.

but for completeness, it would be something like (sketch):

```tsx
const CustomToolbar = () => {
  const handler = useActionHandler(view);
  return (
    <Toolbar.Root onAction={handler} state={getFormatting(view.state)}>
      <Toolbar.Markdown />
      <Toolbar.Separator />
      <Toolbar.Extended />
    </Toolbar.Root>
  );
}

const renderToolbar = (el, view) => {
  createPortal(<CustomToolbar view={view} />, el);
}

const Story = () => {
  return <Editor extensions={[toolbar({ onRender: renderToolbar })]} />
}
```

Reply:
I'll drop this extension, and just make the Toolbar story work with the hook you wrote in the other comment.